src/java/org/apache/cassandra/db/MutationVerbHandler.java

```diff
@@ -61,38 +61,29 @@ private void addSensorsToResponse(Message.Builder<NoPayload> response, Mutation
         // Add write bytes sensors to the response
         Function<String, String> requestParam = SensorsCustomParams::encodeTableInWriteBytesRequestParam;
         Function<String, String> tableParam = SensorsCustomParams::encodeTableInWriteBytesTableParam;
-        Collection<Sensor> sensors = RequestTracker.instance.get().getSensors(Type.WRITE_BYTES);
-        addSensorsToResponse(sensors, requestParam, tableParam, response);
+        Collection<Sensor> requestSensors = RequestTracker.instance.get().getSensors(Type.WRITE_BYTES);
+        addSensorsToResponse(requestSensors, requestParam, tableParam, response);
 
         // Add index write bytes sensors to the response
         requestParam = SensorsCustomParams::encodeTableInIndexWriteBytesRequestParam;
         tableParam = SensorsCustomParams::encodeTableInIndexWriteBytesTableParam;
-        sensors = RequestTracker.instance.get().getSensors(Type.INDEX_WRITE_BYTES);
-        addSensorsToResponse(sensors, requestParam, tableParam, response);
-
-        // Add internode message sensors to the response
-        for (PartitionUpdate update : mutation.getPartitionUpdates())
-        {
-            Context context = Context.from(update.metadata());
-            Optional<Sensor> internodeBytes = SensorsRegistry.instance.getSensor(context, Type.INTERNODE_MSG_BYTES);
-            internodeBytes.ifPresent(sensor -> {
-                byte[] bytes = SensorsCustomParams.sensorValueAsBytes(sensor.getValue());
-                String internodeBytesTableParam = SensorsCustomParams.encodeTableInInternodeBytesTableParam(context.getTable());
-                response.withCustomParam(internodeBytesTableParam, bytes);
-            });
-
-            Optional<Sensor> internodeCount = SensorsRegistry.instance.getSensor(context, Type.INTERNODE_MSG_COUNT);
-            internodeCount.ifPresent(sensor -> {
-                byte[] count = SensorsCustomParams.sensorValueAsBytes(sensor.getValue());
-                String internodeCountTableParam = SensorsCustomParams.encodeTableInInternodeCountTableParam(context.getTable());
-                response.withCustomParam(internodeCountTableParam, count);
-            });
-        }
+        requestSensors = RequestTracker.instance.get().getSensors(Type.INDEX_WRITE_BYTES);
+        addSensorsToResponse(requestSensors, requestParam, tableParam, response);
+
+        // Add internode bytes sensors to the response after updating each per-table sensor with the current response
```

Comment:
I commented in the `ReadCommandVerbHandler` regarding this - I mean if we go with the approximation, we need to make it consistent - here we are including the sensors values from above (`WRITE_BYTES` and `INDEX_WRITE_BYTES`) are already added, but in the read example, `READ_BYTES` is added after we calculate the size.

Reply:
Good point, we can refactor this.